bundles/org.openhab.binding.fronius/src/main/java/org/openhab/binding/fronius/internal/handler/FroniusMeterHandler.java

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.fronius.internal.handler;
+
+import java.util.Map;
+
+import org.apache.commons.lang.StringUtils;
+import org.openhab.binding.fronius.internal.FroniusBaseDeviceConfiguration;
+import org.openhab.binding.fronius.internal.FroniusBindingConstants;
+import org.openhab.binding.fronius.internal.FroniusBridgeConfiguration;
+import org.openhab.binding.fronius.internal.api.MeterRealtimeResponseDTO;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.Thing;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link FroniusMeterHandler} is responsible for updating the data, which are
+ * sent to one of the channels.
+ *
+ * @author Jimmy Tanagra - Initial contribution
+ */
+public class FroniusMeterHandler extends FroniusBaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(FroniusMeterHandler.class);
+    private MeterRealtimeResponseDTO meterRealtimeResponse;
+    private FroniusBaseDeviceConfiguration config;
+
+    public FroniusMeterHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    protected String getDescription() {
+        return "Fronius Smart Meter";
+    }
+
+    @Override
+    public void refresh(FroniusBridgeConfiguration bridgeConfiguration) {
+        updateData(bridgeConfiguration, config);
+        updateChannels();
+        updateProperties();
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(FroniusBaseDeviceConfiguration.class);
+        super.initialize();
+    }
+
+    /**
+     * Update the channel from the last data retrieved
+     *
+     * @param channelId the id identifying the channel to be updated
+     * @return the last retrieved data
+     */
+    @Override
+    protected Object getValue(String channelId) {
+        String[] fields = StringUtils.split(channelId, "#");
+
+        String fieldName = fields[0];
+
+        if (meterRealtimeResponse == null) {
+            return null;
+        }
+        switch (fieldName) {
+            case FroniusBindingConstants.MeterEnable:
+                return meterRealtimeResponse.getBody().getData().getEnable();
```

Comment:
Please cache `meterRealtimeResponse.getBody().getData()` to a local variable and use it instead of calling it a million times.

Reply:
changed as suggested